Sprint-2/step-1

eugenepokalyuk opened the pull request:
Проект. Перенос состояния в Context

eugenepokalyuk committed:
Initial commit

Проект. Перенос состояния в Context

vadmas commented on src/components/BurgerConstructor/BurgerConstructor.js:
```diff
@@ -1,64 +1,113 @@
+    case 'SET_TOTAL_PRICE':
+      return {
+        ...state,
+        totalPrice: action.payload,
```
Можно лучше: для расчета полной стоимости лучше использовать хук useMemo

vadmas commented on src/components/BurgerConstructor/BurgerConstructor.js:
```diff
@@ -1,64 +1,113 @@
+        const data = await fetchIngredientsData();
+        setBurgerIngredients(data);
+      } catch (error) {
+        console.error(error);
```
Можно лучше: оставлять `console.log` в коде после завершения отладки считается плохим тоном

eugenepokalyuk replied:
ой 🫣

vadmas commented on src/components/BurgerConstructor/BurgerConstructor.js:
```diff
@@ -1,64 +1,113 @@
       <div>
-        <div className={styles.flexContainer} style={{ display: 'flex', flexDirection: 'column', gap: '10px' }}>
-          <ConstructorIngredients items={BasketData} />
+        <div className={styles.flexContainer} style={{ display: 'flex', flexDirection: 'column', gap: '10px', alignItems: 'flex-end' }}>
```
Нужно исправить: не стоит использовать инлайновые стили. Целесообразнее реализовывать описание стилей единообразно в файлах .module.scss

vadmas commented on src/components/BurgerConstructor/BurgerConstructor.js:
```diff
@@ -1,64 +1,113 @@
         </div>
       </div>
-
       <Modal isOpen={isModalOpen} onClose={closeModal}>
```
Нужно исправить: для управления отображением модального окна следует воспользоваться условным рендерингом ( open && <Modal … ), а не передавать флаг открытия в компонент модального окна.

vadmas commented on src/components/ConstructorIngredients/ConstructorIngredients.js:
```diff
@@ -1,62 +1,83 @@
-  const firstBunPrice = 20; // Стоимость первой булки
-  const lastBunPrice = 20; // Стоимость последней булки
+const ConstructorIngredients = ({ items, onIngredientAdd, onIngredientRemove }) => {
+  const { burgerIngredients } = useContext(BurgerContext);
```
Отлично: правильно используется хук useContext

vadmas commented on src/components/ConstructorIngredients/ConstructorIngredients.js:
```diff
@@ -1,62 +1,83 @@
-            </div>
-          ))}
+  const calculateTotalPrice = () => {
+    const nonBunIngredients = burgerIngredients.filter((item) => item.type !== 'bun');
```
Можно лучше: переиспользуемые строковые значения типа bun, main, sauses – лучше выносить в константы

vadmas commented on src/services/BurgerContext.js:
```diff
@@ -0,0 +1,33 @@
+  };
+
+  return (
+    <BurgerContext.Provider
```
Отлично: правильно реализована работа с контекстом

vadmas reviewed (commented):
Хорошая работа. 
Приложение приобретает готовый вид, видно как вы прогрессируете по мере прохождения курса.

В положительную сторону хочется отметить:

- Правильно реализована работа с Context 
- Корректно работают запросы к внешнему API
- Хорошо осуществлено оформление заказа

**Но есть несколько замечаний, оставленных в коде, которые необходимо будет исправить**

**Места, в которых можно улучшить проект:**
- Вычисления, фильтрацию и маппинг данных, выполняемых на стороне компонента, целесообразно оборачивать в хук useMemo. Так повторные пересчеты значений не будут выполняться на каждый перерендер компонента, а только при изменении установленных зависимостей. Подробнее о данном хуке можно почитать здесь: https://ru.reactjs.org/docs/hooks-reference.html#usememo

Пожалуйста, учтите выше описанные замечания и те, что оставлены коде вашей работы (особенно с пометкой "нужно исправить").
При повторной проверке, если все замечания будут исправлены, работа будет принята.

Удачного рефакторинга кода!

eugenepokalyuk committed:
code refactoring

1. to calculate the total cost, I used the useMemo hook
2. Removed all console.logs from the code
3. Removed all inline styles from the code
4. To control the display of the modal window you should use conditional rendering
5. Put reusable string values into constants: bun, main, sause

eugenepokalyuk committed:
Update BurgerConstructor.js

Remove old code, Ooops...

eugenepokalyuk committed:
Update BurgerConstructor.js

replace reducer to useMemo

vadmas commented:
Отлично, все критичные комментарии исправлены. Работу принимаю. Удачи в дальнейшем обучении!
